internal/kafka/test/integration/cluster_deregistration_endpoint_test.go

```diff
@@ -123,49 +121,41 @@ func TestEnterpriseClusterDeregistration(t *testing.T) {
 	closeRespBody(resp)
 
 	// no org id account deregistration non enterprise cluster
-	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(nonadminCtx, true, id1, &forceTrue)
+	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(nonadminCtx, true, id1)
 	g.Expect(e).To(gomega.HaveOccurred(), "error should be thrown when attempting to call deregister endpoint with no org ID in the token")
 	closeRespBody(resp)
 
 	// not found cluster deregistration non enterprise cluster
-	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(adminCtx, true, nonExistentID, &forceTrue)
+	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(adminCtx, true, nonExistentID)
 	g.Expect(e).To(gomega.HaveOccurred(), "error should be thrown when attempting to call deregister endpoint against non-existent cluster")
 	closeRespBody(resp)
 
 	// other org cluster deregistration non enterprise cluster
-	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(otherAdminadminCtx, true, id1, &forceTrue)
+	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(otherAdminadminCtx, true, id1)
 	g.Expect(e).To(gomega.HaveOccurred(), "error should be thrown when attempting to deregister enterprise cluster from another org")
 	closeRespBody(resp)
 
 	// non enterprise cluster deregistration non enterprise cluster
-	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(adminCtx, true, id3, &forceTrue)
+	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(adminCtx, true, id3)
 	g.Expect(e).To(gomega.HaveOccurred(), "error should be thrown when attempting to deregister non-enterprise cluster")
 	closeRespBody(resp)
 
-	// force false with kafkas on the cluster failure
-	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(adminCtx, true, id1, &forceFalse)
-	g.Expect(e).To(gomega.HaveOccurred(), "error should be thrown when attempting to deregister enterprise cluster with kafkas on it without force=true")
+	// should fail when kafkas are on the cluster
+	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(adminCtx, true, id1)
+	g.Expect(e).To(gomega.HaveOccurred(), "error should be thrown when attempting to deregister enterprise cluster with kafkas on it")
 	closeRespBody(resp)
 
-	// force false with empty cluster success
-	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(adminCtx, true, id2, &forceFalse)
+	// should succeed with empty cluster
+	_, resp, e = client.EnterpriseDataplaneClustersApi.DeleteEnterpriseClusterById(adminCtx, true, id2)
 	g.Expect(e).NotTo(gomega.HaveOccurred(), "error should not be thrown when attempting to deregister enterprise cluster with no kafkas on it without force=true")
```

Comment:
I've two more suggestions in regard to this test:

1. We could add another assertion here that attempts to register the same cluster that was successfully deleted. To test that cluster can be re-registered successfully. For that to work, we'll need to wait first for the cluster to be fully removed - with something like https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5545b76d18126733b9648f2c4855430567012c08/internal/kafka/test/integration/cluster_mgr_test.go#L346 

And just after that, we attempt to re-register the cluster agan.

2. It has to be run only against for `integration` env. If run against `development` env it'll fail as it'll attempt to delete an syncset from a dummy cluster that does not exist in OCM

This is a general comment the whole test https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5545b76d18126733b9648f2c4855430567012c08/internal/kafka/test/integration/cluster_deregistration_endpoint_test.go#L19 

Something like https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/5545b76d18126733b9648f2c4855430567012c08/internal/kafka/test/integration/cluster_get_enterprise_test.go#L32-L35 can be used.